modules/redhen_contact/redhen_contact.module

```diff
@@ -1081,6 +1093,34 @@ function redhen_contact_form_user_register_form_alter(&$form, &$form_state) {
 }
 
 /**
+ * Helper function to embed a contact form on a user form.
+ *
+ * @param array $form
+ *   Form array.
+ * @param array $form_state
+ *   Form state array.
+ * @param RedhenContact $contact
+ *   The contact to build the form on.
+ * @param string $submit_handler
+ *   Function name of the submit handler to add to the form.
+ */
+function _redhen_contact_user_embed_contact_form(&$form, &$form_state, RedhenContact $contact, $submit_handler) {
```

Comment:
It feels strange to have an argument for the submit handler. This function shouldn't care... And besides, you have access to the form itself; just add it yourself in a separate line:
```php
_redhen_contact_user_embed_contact_form($form, $form_state, $contact_object);
$form['#submit'][] = 'redhen_contact_user_registration_submit';
```